commandapi-platforms/commandapi-velocity/commandapi-velocity-plugin/src/main/java/dev/jorel/commandapi/config/VelocityConfigurationAdapter.java

```diff
@@ -0,0 +1,105 @@
+package dev.jorel.commandapi.config;
+
+import org.spongepowered.configurate.CommentedConfigurationNode;
+import org.spongepowered.configurate.ConfigurationNode;
+import org.spongepowered.configurate.serialize.SerializationException;
+import org.spongepowered.configurate.yaml.YamlConfigurationLoader;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+@SuppressWarnings("ConfusingArgumentToVarargsMethod")
+public record VelocityConfigurationAdapter(YamlConfigurationLoader loader, CommentedConfigurationNode config) implements ConfigurationAdapter<ConfigurationNode, DefaultedVelocityConfig> {
+
+	@Override
+	public void setValue(String key, Object value) {
+		setValue0(key, value);
+	}
+
+	@Override
+	public void setComment(String key, String[] comment) {
+		config.node(key.split("\\.")).comment(String.join("\n", comment));
```

Comment:
I see the code `config.node(key.split("\\."))` repeated a lot. Should it be moved to its own method?

```java
@SuppressWarnings("ConfusingArgumentToVarargsMethod")
public CommentedConfigurationNode getNode(String path) {
	return config.node(path.split("\\."));
}
```

That would also mean that `ConfusingArgumentToVarargsMethod` would only need to be suppressed in one location (though I would like to figure out how to resolve that warning rather than suppressing it).